app/components/sidebars/main/channels_list/list/index.js

```diff
@@ -2,46 +2,69 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
+import {createSelector} from 'reselect';
 
 import {General} from 'mattermost-redux/constants';
 import {
     getSortedUnreadChannelIds,
-    getSortedFavoriteChannelIds,
-    getSortedPublicChannelIds,
-    getSortedPrivateChannelIds,
-    getSortedDirectChannelIds,
+    getOrderedChannelIds,
 } from 'mattermost-redux/selectors/entities/channels';
 import {getCurrentUserId, getCurrentUserRoles} from 'mattermost-redux/selectors/entities/users';
 import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 import {getTheme, getFavoritesPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {getSidebarPreferences} from 'mattermost-redux/selectors/entities/sidebar';
 import {showCreateOption} from 'mattermost-redux/utils/channel_utils';
 import {isAdmin as checkIsAdmin, isSystemAdmin as checkIsSystemAdmin} from 'mattermost-redux/utils/user_utils';
 import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
 
+import {SidebarSectionTypes} from 'app/constants/view';
+
 import List from './list';
 
+const filterZeroUnreads = createSelector(
```

Comment:
can this selector be moved to `app/selectors/sidebar` or something and not just place it here

Reply:
We could consider changing that policy for very specific selectors like this since it could be nice to keep those with the component itself